.github/workflows/optimizer.yml

```diff
@@ -0,0 +1,78 @@
+name: optimizer test
+
+on:
+  push:
+    branches:
+      - "main"
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+
+  pull_request:
+    branches:
+      - "main"
+  schedule:
+    # Trigger test every day at 00:03 clock UTC
+    - cron: "3 0 * * *"
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  run_optimizer:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.19.6"
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: cache go mod
+        uses: actions/cache@v3
```

Comment:
Should also add the cache for cargo build?

Reply:
Good point. Already added it.